src/Extensions/AzureBlobPayloads/PayloadStore/BlobPayloadStore.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Globalization;
+using System.IO.Compression;
+using System.Text;
+using Azure;
+using Azure.Core;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.DurableTask.Converters;
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Azure Blob Storage implementation of <see cref="IPayloadStore"/>.
+/// Stores payloads as blobs and returns opaque tokens in the form "blob:v1:&lt;container&gt;:&lt;blobName&gt;".
+/// </summary>
+internal sealed class BlobPayloadStore : IPayloadStore
+{
+    const string TokenPrefix = "blob:v1:";
+    const string ContentEncodingGzip = "gzip";
+    const int DefaultCopyBufferSize = 81920;
+    const int MaxRetryAttempts = 8;
+    const int BaseDelayMs = 250;
+    const int MaxDelayMs = 10_000;
+    const int MaxJitterMs = 100;
+    const int NetworkTimeoutMinutes = 2;
+
+    // Jitter RNG for retry backoff
+    static readonly object RandomLock = new object();
+    static readonly Random SharedRandom = new Random();
+    readonly BlobContainerClient containerClient;
+    readonly LargePayloadStorageOptions options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BlobPayloadStore"/> class.
+    /// </summary>
+    /// <param name="options">The options for the blob payload store.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="options"/> is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when <paramref name="options.ConnectionString"/> is null or empty.</exception>
+    public BlobPayloadStore(LargePayloadStorageOptions options)
+    {
+        this.options = options ?? throw new ArgumentNullException(nameof(options));
+
+        Check.NotNullOrEmpty(options.ConnectionString, nameof(options.ConnectionString));
+        Check.NotNullOrEmpty(options.ContainerName, nameof(options.ContainerName));
+
+        BlobClientOptions clientOptions = new()
+        {
+            Retry =
+            {
+                Mode = RetryMode.Exponential,
+                MaxRetries = MaxRetryAttempts,
+                Delay = TimeSpan.FromMilliseconds(BaseDelayMs),
+                MaxDelay = TimeSpan.FromSeconds(MaxDelayMs),
+                NetworkTimeout = TimeSpan.FromMinutes(NetworkTimeoutMinutes),
+            },
+        };
+        BlobServiceClient serviceClient = new(options.ConnectionString, clientOptions);
+        this.containerClient = serviceClient.GetBlobContainerClient(options.ContainerName);
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> UploadAsync(ReadOnlyMemory<byte> payloadBytes, CancellationToken cancellationToken)
+    {
+        // One blob per payload using GUID-based name for uniqueness (stable across retries)
+        string timestamp = DateTimeOffset.UtcNow.ToString("yyyy/MM/dd/HH/mm/ss", CultureInfo.InvariantCulture);
```

Comment:
I'm not a huge fan of this naming scheme. We might want to make it configurable so customers can control if it overlaps with their own naming schemes. Regardless, I think we may want to put it under a DTS indicated prefix like `dts/payloads` or something similar. Then I'd just a GUID or hash for the name. Since this should be stored in the input/output of the activities, it doesn't need to be super human readable.

However, this is coming from my own background so if people don't agree we can keep it this way.

Reply:
Also, if we rerun this step a lot, will it upload a bunch of new files? It'd be good if it was idempotent to the same key.